names/names.go

```diff
@@ -0,0 +1,344 @@
+		{"Az", "AZ", "az", nil},
+		{"Bgp", "BGP", "bgp", nil},
+		{"Cors", "CORS", "cors", nil},
+		{"Cidr", "CIDR", "cidr", nil},
+		{"Cname", "CNAME", "cname", nil},
+		{"Cpu", "CPU", "cpu", nil},
+		{"Crl", "CRL", "crl", nil},
+		{"Cps", "CPS", "cps", nil},
+		{"Csr", "CSR", "csr", nil},
+		{"Dhcp", "DHCP", "dhcp", nil},
+		{"Dns", "DNS", "dns", nil},
+		{"Dpd", "DPD", "dpd", nil},
+		{"Ebs", "EBS", "ebs", nil},
+		{"Ec2", "EC2", "ec2", nil},
+		{"Ecr", "ECR", "ecr", nil},
+		// Prevent "Edit" from becoming "EDIt"
+		{"Edi", "EDI", "edi", re2.MustCompile("Edi(?!t)", re2.None)},
+		{"Efs", "EFS", "efs", nil},
+		{"Eks", "EKS", "eks", nil},
+		// Prevent "Enable" and "Enabling" from becoming "ENAble"
+		{"Ena", "ENA", "ena", re2.MustCompile("Ena(?!bl)", re2.None)},
+		{"Ecmp", "ECMP", "ecmp", nil},
+		{"Fpga", "FPGA", "fpga", nil},
+		{"Gpu", "GPU", "gpu", nil},
+		{"Html", "HTML", "html", nil},
+		{"Http", "HTTP", "http", nil},
+		{"Https", "HTTPS", "https", nil},
+		{"Iam", "IAM", "iam", nil},
+		{"Icmp", "ICMP", "icmp", nil},
+		// Prevent "IOPS" from becoming "IOps"
+		{"Io", "IO", "io", re2.MustCompile("Io(?!ps)", re2.None)},
+		{"Iops", "IOPS", "iops", nil},
+		{"Json", "JSON", "json", nil},
+		{"Jwt", "JWT", "jwt", nil},
+		{"Kms", "KMS", "kms", nil},
+		{"Ldap", "LDAP", "ldap", nil},
+		{"Mfa", "MFA", "mfa", nil},
+		// Prevent "Native" from becoming "NATive"
+		{"Nat", "NAT", "nat", re2.MustCompile("Nat(?!i)", re2.None)},
+		{"Oidc", "OIDC", "oidc", nil},
+		{"Ocsp", "OCSP", "ocsp", nil},
+		// Capitalize the 'd' following RAM in certain cases
+		{"Ramdisk", "RAMDisk", "ramDisk", re2.MustCompile("Ramdisk", re2.None)},
+		// Model fields starting with 'Ram' refer to RAM
+		{"Ram", "RAM", "ram", re2.MustCompile("Ram", re2.None)},
+		{"Rfc", "RFC", "rfc", nil},
+		{"Sdk", "SDK", "sdk", nil},
+		{"Sha256", "SHA256", "sha256", nil},
+		{"Sns", "SNS", "sns", nil},
+		{"Sqs", "SQS", "sqs", nil},
+		{"Sriov", "SRIOV", "sriov", nil},
+		{"Sse", "SSE", "sse", nil},
+		{"Ssl", "SSL", "ssl", nil},
+		{"Tcp", "TCP", "tcp", nil},
+		{"Tde", "TDE", "tde", nil},
+		{"Tls", "TLS", "tls", nil},
+		{"Udp", "UDP", "udp", nil},
+		// Need to prevent "security" from becoming "SecURIty"
+		{"Uri", "URI", "uri", re2.MustCompile("(?!sec)uri(?!ty)|(Uri)", re2.None)},
+		{"Url", "URL", "url", nil},
+		{"Uuid", "UUID", "uuid", nil},
+		{"Vlan", "VLAN", "vlan", nil},
+		{"Vpc", "VPC", "vpc", nil},
+		{"Vpn", "VPN", "vpn", nil},
+		{"Vgw", "VGW", "vgw", nil},
+		{"Waf", "WAF", "waf", nil},
+		{"Xml", "XML", "xml", nil},
+		{"Yaml", "YAML", "yaml", nil},
+	}
+)
+
+var goKeywords = []string{
+	"break",
+	"case",
+	"chan",
+	"const",
+	"continue",
+	"default",
+	"defer",
+	"else",
+	"fallthrough",
+	"for",
+	"func",
+	"go",
+	"goto",
+	"if",
+	"import",
+	"interface",
+	"map",
+	"package",
+	"range",
+	"return",
+	"select",
+	"struct",
+	"switch",
+	"type",
+	"var",
+}
+
+// Names contains variations of a name
+type Names struct {
+	Original      string
```

Comment:
@jaypipes i just realized that `ModelOriginal` no longer exists. Is there a specific reason for that? `ModelOriginal` was used in `fields.go` and `model.go` https://github.com/aws-controllers-k8s/code-generator/search?q=modeloriginal

Reply:
@A-Hilaly Yes, I wanted to cleanly separate out code-generator-specific model code (like ModelOriginal) from the Names struct. ModelOriginal belongs on the code-generator/pkg/model.CRD field.